signal_scanner_bot/env.py

```diff
@@ -85,7 +117,10 @@ def log_vars() -> None:
 SIGNAL_MESSAGE_HEADERS = _env(
     "SIGNAL_MESSAGE_HEADERS", convert=_cast_to_set, default={}
 )
+AUTOSCAN_STATE_FILE_PATH = _env(
+    "AUTOSCAN_STATE_FILE_PATH", convert=_cast_to_string, default=""
+)
 
 SIGNAL_LOCK = Lock()
 
-STATE = _State()
+STATE = _State(Path(AUTOSCAN_STATE_FILE_PATH)) if AUTOSCAN_STATE_FILE_PATH else _State()
```

Comment:
As an alternative to this logic, perhaps you could define the default which is currently in the `_State` init on L121 instead, then just have this last line be `STATE = _State(AUTOSCAN_STATE_FILE_PATH)`. You could even add `Path` as the convert method to L120:
```
AUTOSCAN_STATE_FILE_PATH = _env("AUTOSCAN_STATE_FILE_PATH", conver=Path, default="signal_scanner_bot/.autoscanner-state-file")
```